### DISCUSSION TITLE
Calling a C++ UDF from python API

### DISCUSSION THREAD

**Discussion by mpage-uga** [duckdb/duckdb#2070] [Q&A]
I have written the `bigger_than_four()` UDF function below, given in the doc.
I am able to run it from C++ (see below).
Is there a way to run `bigger_than_four()` UDF function from the python API ?

```
#include "duckdb.hpp"

using namespace duckdb;

bool bigger_than_four(int value) {
    return value > 4;
}

int main() {
	DuckDB db(nullptr);
	Connection con(db);

	con.CreateScalarFunction<bool, int>("bigger_than_four", &bigger_than_four);

	con.Query("CREATE TABLE integers(i INTEGER)");
	con.Query("INSERT INTO integers VALUES (3)");
	con.Query("INSERT INTO integers VALUES (4)");
	con.Query("INSERT INTO integers VALUES (5)");	

	auto result = con.Query("SELECT bigger_than_four(i) FROM integers");
	result->Print();
}
```

**Comment by Mytherin** [comment]
To use a C++ UDF you will need to make a loadable extension, see e.g. [here](https://github.com/duckdb/duckdb/tree/master/test/extension), but this is not very well documented yet. We also have some support for Python UDFs, perhaps @hannesmuehleisen can expand on that.

**Comment by mpage-uga** [comment]
I have built the whole DuckDB stuff using `Makefile` in `duckdb` directory, and then tried to build the `loadable_extension_demo.cpp` extension in duckdb/test/extension, but when I run the Makefile I have built for this (there are none in this directory), I get:

```
 [100%] Linking CXX shared library loadable_extension_demo.duckdb_extension
/usr/bin/ld : couldn't find -lduckdb
```
Should I add a specific `link_directories` entry in CMakeLists.txt ?
